tests/foreman/api/test_product.py

```diff
@@ -367,3 +369,31 @@ def test_positive_sync_several_repos(self):
         self.assertGreaterEqual(rpm_repo.read().content_counts['rpm'], 1)
         self.assertGreaterEqual(
             puppet_repo.read().content_counts['puppet_module'], 1)
+
+    @tier2
+    def test_positive_filter_product_list(self):
+        """ Filter products based on param 'custom/redhat_only'
+
+        :id: e61fb63a-4552-4915-b13d-23ab80138249
+
+        :expectedresults: Able to list the products based on defined filter.
+
+        :CaseLevel: Integration
+
+        :BZ: 1667129
+        """
+        product = entities.Product(organization=self.org.id).create()
+        # Manifest upload to create RH Product
+        with manifests.clone() as manifest:
+            upload_manifest(self.org.id, manifest.content)
+
+        custom_products = entities.Product(organization=self.org.id).search(query={'custom': True})
+        rh_products = entities.Product(organization=self.org.id).search(
+            query={'redhat_only': True})
+
+        self.assertEqual(len(custom_products), 1)
+        self.assertIn(product.name, [prod.name for prod in custom_products])
```

Comment:
why need to create list comprehension as there will only 1 custom_product ? as you already have added check above statement.
`
self.assertEqual(product.name, custom_products[0].name)`

Reply:
yeah we not needed list comprehension here .......nice catch ...thanks

Reply:
it's update now . revisit please.